apis/r/R/utils-assertions.R

```diff
@@ -185,3 +185,30 @@ recursively_make_integer64 <- function(x) {
     }
     x
 }
+
+#' Warn if using a SOMADenseNDArray
+#' @param collection_name name of the SOMA collection
+#' @param layer a SOMASparseNDArray or SOMADenseNDArray
+#' @return Invisible logical indicating if the layer is sparse
+#' @references https://github.com/single-cell-data/TileDB-SOMA/issues/1245
+#' @noRd
+
+warn_if_dense <- function(collection_name, layer) {
+  stopifnot(
+    is_scalar_character(collection_name),
+    inherits(layer, "SOMASparseNDArray") || inherits(layer, "SOMADenseNDArray")
+  )
+  is_sparse <- inherits(layer, "SOMASparseNDArray")
+  if (!is_sparse) {
+    msg1 <- sprintf(
+      "Values for '%s' are encoded as dense instead of sparse\n",
+      basename(layer$uri)
+    )
+    msg2 <- sprintf(
+      "  - all '%s' arrays should be saved as 'SOMASparseNDArrays'\n",
+      collection_name
+    )
+    warning(msg1, msg2, call. = FALSE, .immediate = TRUE)
```

Comment:
The warning exists to inform the user that their SOMA is techinically valid, but uses data structures that are not preferred. I belive this information is useful, but you're right that doing it every time is a bit much. We could set it to once-per-session using R options

```suggestion
    if (isTRUE(getOption("tiledbsoma.warnings.m_axis_dense", TRUE))) {
      warning(msg1, msg2, call. = FALSE, immediate.= TRUE)
      options(tiledbsoma.warnings.m_axis_dense = FALSE)
    }
```

Reply:
(note that this will likely require changes to the tests)

Reply:
I like the idea of once per-session warnings. There are a few places in the package we could apply this to so let's punt for now and make a separate PR that implements it package-wide.